xarray/core/treenode.py

```diff
@@ -663,11 +671,13 @@ def name(self) -> str | None:
 
     @name.setter
     def name(self, name: str | None) -> None:
-        if name is not None:
-            if not isinstance(name, str):
-                raise TypeError("node name must be a string or None")
-            if "/" in name:
-                raise ValueError("node names cannot contain forward slashes")
+        if self.parent is not None:
+            raise ValueError(
+                "cannot set the name of a node with a parent. Consider creating "
```

Comment:
(Slightly clearer wording)

```suggestion
                "cannot set the name of a node which already has a parent. Consider creating "
```